lib/chef/resource.rb

```diff
@@ -55,6 +55,8 @@ class Resource
     # This lets user code do things like `not_if { shell_out!("command") }`
     include Chef::Mixin::ShellOut
 
+    NULL_ARG = Object.new
```

Comment:
Mostly I am not converting over old things.  Main reason to do this is it lets you distinguish between "setting something to nil" and "getting the value."